### PR TITLE
chore(localdebug): add spfx v1.16 node checker

### PR DESCRIPTION
**Pull request by XiaofuHuang**
1. add spfx v1.16 node checker
2. remove func node checker
3. Scaffold transparency tasks for SPFx project
 
image

### CODE REVIEW

File: packages/vscode-extension/src/handlers.ts [hunk 1]
```diff
@@ -1644,22 +1644,8 @@ export async function openReadMeHandler(args: any[]) {
 
 export async function promptSPFxUpgrade() {
   if (globalVariables.isSPFxProject) {
-    let projectSPFxVersion = null;
     // eslint-disable-next-line @typescript-eslint/no-non-null-assertion, @typescript-eslint/no-non-null-asserted-optional-chain
-    const yoInfoPath = path.join(globalVariables.workspaceUri?.fsPath!, "SPFx", ".yo-rc.json");
```
Comment (swatDong) [inline packages/vscode-extension/src/handlers.ts:1649]: Any compatibility issue if removing this? E.g., opening old project?

Comment (HuihuiWu-Microsoft) [reply packages/vscode-extension/src/handlers.ts:1649]: The logic doesn't change, I just moved it to util function so prerequisite checker can reuse.

**Review by swatDong — DISMISSED** [review]

File: packages/fx-core/src/common/local/localEnvManager.ts [hunk 2]
```diff
@@ -59,9 +62,11 @@ export class LocalEnvManager {
 
     // NodeJS
     if (isSPFx) {
-      depsTypes.push(DepsType.SpfxNode);
-    } else if (includeBackend || includeFuncHostedBot) {
-      depsTypes.push(DepsType.FunctionNode);
+      if ((await getAppSPFxVersion(projectPath))?.startsWith("1.16.0")) {
```
Comment (tecton) [inline packages/fx-core/src/common/local/localEnvManager.ts:65]: Does it mean the supported version in future is 1.16.0-x instead of 1.16.x?

**Review by tecton — COMMENTED** [review]
In the screenshot, it seems that the error message should be used in VS Code prompt is printed because there's a markdown link. If not related to this PR, we can fix it later.

File: packages/fx-core/src/common/local/localEnvManager.ts (hunk 2 above)
Comment (XiaofuHuang) [reply packages/fx-core/src/common/local/localEnvManager.ts:65]: @HuihuiWu-Microsoft  As we discussed offline, after SPFx 1.16.0 GA, Teams Toolkit will only support node.js 16. The old SPFx node checker will be retired.

Comment (HuihuiWu-Microsoft) [reply packages/fx-core/src/common/local/localEnvManager.ts:65]: Yes, for this release, we'll support SPFx v1.16.0-beta.1

**Review by tecton — APPROVED** [review]

File: packages/fx-core/src/common/local/localEnvManager.ts (hunk 2 above)
Comment (HuihuiWu-Microsoft) [reply packages/fx-core/src/common/local/localEnvManager.ts:65]: @XiaofuHuang Right, once SPFx v1.16.0 released, we'll only support SPFx v1.16.0(Node.js v16 is supported by SPFx v1.16.0) and there'll be only one node checker for SPFx

**Review by swatDong — APPROVED** [review]

**Review by jayzhang — APPROVED** [review]

**Pull request merged**